lib/f_core/protocols/horus/horus.c

```diff
@@ -0,0 +1,817 @@
+                outbyte |= outbit;
+                noutbits++;
+                if (noutbits % 8) {
+                    outbyte <<= 1;
+                } else {
+                    LOG_INF("output payload byte: 0x%02x", outbyte);
+                    *pout++ = outbyte;
+                    outbyte = 0;
+                }
+            }
+
+            ingolay = 0;
+        }
+    } /* while(.... */
+
+    LOG_INF("finishing up .....");
+
+    /* Complete final Golay decode  */
+
+    int golayparity = 0;
+    if (ningolay % 12) {
+        for (i = 0; i < 11; i++) {
+            shift = 7 - (nparitybits % 8);
+            golayparitybit = (paritybyte >> shift) & 0x1;
+            golayparity |= golayparitybit;
+            if (i != 10) golayparity <<= 1;
+            nparitybits++;
+            if ((nparitybits % 8) == 0) {
+                /* OK grab a new byte */
+                paritybyte = *pin++;
+                LOG_INF("  Read paritybyte: 0x%02x", paritybyte);
+            }
+        }
+
+        ingolay >>= 1;
+        int codeword = (ingolay << 12) + golayparity;
+        LOG_INF("  ningolay: %d ingolay: 0x%04x", ningolay, ingolay);
+        LOG_INF("  golay code word: 0x%04x", codeword);
+        LOG_INF("  golay decode...: 0x%04x", golay23_decode(codeword));
+
+        outdata = golay23_decode(codeword) >> 11;
+        LOG_INF("  outdata...: 0x%04x", outdata);
+        LOG_INF("  num_payload_data_bits: %d noutbits: %d", num_payload_data_bits, noutbits);
+
+        /* write final byte */
+
+        int ntogo = num_payload_data_bits - noutbits;
+        for (i = 0; i < ntogo; i++) {
+            shift = ntogo - i;
+            outbit = (outdata >> shift) & 0x1;
+            outbyte |= outbit;
+            noutbits++;
+            if (noutbits % 8) {
+                outbyte <<= 1;
+            } else {
+                LOG_INF("  output payload byte: 0x%02x", outbyte);
+                *pout++ = outbyte;
+                outbyte = 0;
+            }
+        }
+    }
+
+    LOG_INF("pin - output_payload_data: %ld num_payload_data_bytes: %d", pout - output_payload_data,
+            num_payload_data_bytes);
+
+    assert(pout == (output_payload_data + num_payload_data_bytes));
+}
+#endif
+
+#ifdef CONFIG_HORUS_INTERLEAVER
+
+uint16_t primes[] = {2,   3,   5,   7,   11,  13,  17,  19,  23,  29,  31,  37,  41,  43,  47,  53,  59,  61,  67,  71,
+                     73,  79,  83,  89,  97,  101, 103, 107, 109, 113, 127, 131, 137, 139, 149, 151, 157, 163, 167, 173,
+                     179, 181, 191, 193, 197, 199, 211, 223, 227, 229, 233, 239, 241, 251, 257, 263, 269, 271, 277, 281,
+                     283, 293, 307, 311, 313, 317, 331, 337, 347, 349, 379, 383, 389, 757, 761, 769, 773};
+
+void interleave(unsigned char *inout, int nbytes, int dir) {
+    uint16_t nbits = (uint16_t) nbytes * 8;
+    uint32_t i, j, n, ibit, ibyte, ishift, jbyte, jshift;
+    uint32_t b;
+    unsigned char out[nbytes];
+
+    memset(out, 0, nbytes);
```

Comment:
```suggestion
    unsigned char out[nbytes] = {0};
```

Reply:
#not_my_software_lead that code doesnt compile cuz thats a VLA. `out[nbytes] = {}` is valid C23 tho and does the same thing

Reply:
Imagine using VLAs

Reply:
Be omnipotent and know how many bytes your arrays are at compile time